src/payment.js

```diff
@@ -0,0 +1,188 @@
+const cartApi = require('./cart');
+const cardApi = require('./card');
+const { isFunction, vaultRequest } = require('./utils');
+
+function methods(request) {
+  return {
+    async createElements(params = {}) {
+      const cart = await cartApi.methods(request).get();
+      if (!cart) {
+        throw new Error('Cart not found');
+      }
+      const gateways = await vaultRequest('get', '/gateways');
+      if (gateways.error) {
+        throw new Error(gateways.error);
+      }
+      await render(request, cart, gateways, params);
+    },
+  };
+}
+
+async function render(request, cart, gateways, params) {
+  if (gateways.braintree && gateways.paypal) {
+    if (!(window.braintree && window.braintree.client)) {
+      await loadScript(
+        'braintree-web',
+        'https://js.braintreegateway.com/web/3.57.0/js/client.min.js',
+      );
+    }
+    if (!(window.braintree && window.braintree.paypalCheckout)) {
+      await loadScript(
+        'braintree-web-paypal-checkout',
+        'https://js.braintreegateway.com/web/3.57.0/js/paypal-checkout.min.js',
+      );
+    }
+    if (!window.paypal) {
+      await loadScript(
+        'paypal-sdk',
+        `https://www.paypal.com/sdk/js?client-id=${gateways.paypal.client_id}&merchant-id=${gateways.paypal.merchant_id}&vault=true`,
+      );
+    }
+    return await braintreePayPalButton(request, cart, gateways, params);
+  } else if (gateways.stripe) {
+    if (!window.Stripe) {
+      await loadScript('stripe-js', 'https://js.stripe.com/v3/');
+    }
+    return await stripeElements(request, gateways, params);
+  }
+
+  throw new Error('Gateway elements are not implemented');
+}
+
+const loading = {};
+
+const loadScript = async (id, src) =>
+  !loading[id] &&
+  new Promise((resolve) => {
```

Comment:
@vstudenichnik-insoft  Small problem here: if this is called twice quickly, the 2nd time it will return false even if the promise from the 1st call isn't resolve yet, and then will continue to `await stripeElements` even if it hasn't been loaded yet.

Instead it should return a promise that gets resolved once the script is loaded even if it's already in the process of loading.

Reply:
Done